apis/logging/v1/input_source_types.go

```diff
@@ -0,0 +1,52 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// NOTE: The Enum validation on SourceSpec.Type must be updated if the list of types changes.
+
+// SourceSpec is a union of input source types.
+//
+// The fields of this struct define the set of known source types.
+type SourceSpec struct {
+	HTTP *HTTPSource `json:"http,omitempty"`
+}
+
+const (
+	FormatK8S  = "k8s"  // Log events in k8s list format, e.g. API audit log events.
+	FormatText = "text" // Newline-separated plain-text log records.
+)
+
+// HTTPSource receives encoded logs as a HTTP endpoint.
+type HTTPSource struct {
```

Comment:
Do we need to be able to support auth of some kind (i.e. certificates, token).  If so maybe it goes up a level on source

Reply:
Yes - thats To Be Done, see FIXME comment below;